readthedocs/projects/forms.py

```diff
@@ -1025,8 +1026,35 @@ def clean_domain(self):
             if invalid_domain and domain_string.endswith(invalid_domain):
                 raise forms.ValidationError(f"{invalid_domain} is not a valid domain.")
 
+        if self._has_cname(domain_string):
+            raise forms.ValidationError(
+                _(
+                    "This domain has a CNAME record. "
+                    "Please remove it before adding the domain.",
+                ),
+            )
+
         return domain_string
 
+    def _has_cname(self, domain):
+        """
+        Check if a domain has a CNAME record.
+
+        We make use of `dig` to check if a domain has a CNAME record,
+        if it doesn't return an empty string, it means it has a CNAME record.
+        """
+        try:
+            result = subprocess.run(
+                ["dig", "-t", "CNAME", "+short", domain],
```

Comment:
Are we shelling out to dig for a specific reason instead of using dnspython for any specific reason?

Reply:
We no longer have dnspython in our deps, but we also need to install dnsutils in the docker images, so, probably just depending on dnspython should be fine.